src/app/services/constraint.service.ts

```diff
@@ -579,40 +641,31 @@ export class ConstraintService {
           }
         });
         this.variablesUpdated.next(this.variables);
+        this.treeNodeService
+          .updateVariablesTreeData(
+            this.selectedStudyConceptCountMap,
+            this.selectedConceptCountMap,
+            this.selectedStudyCountMap);
         this.isUpdatingVariables = false;
         resolve(true);
       }
     });
   }
 
-  updateVariablesTreeData() {
-    this.treeNodeService.updateVariablesTreeData(this.selectedStudyConceptCountMap,
-      this.selectedConceptCountMap,
-      this.selectedStudyCountMap);
-  }
-
   importVariablesByNames(names: string[]) {
-    if (this.variablesViewMode === VariablesViewMode.TREE_VIEW) {
-      return this.treeNodeService.selectVariablesTreeNodesByNames(this.treeNodeService.variablesTreeData, names);
-    } else if (this.variablesViewMode === VariablesViewMode.CATEGORIZED_VIEW) {
-      return this.selectVariablesByProperty(names, 'name');
-    }
+    // update the selected tree nodes in gb-variables
+    this.treeNodeService.selectVariablesTreeDataByFields(
+      this.treeNodeService.variablesTreeData, names, ['metadata', 'item_name']);
+    // dispatch the event telling its subscribers that the selected tree nodes have been updated
```

Comment:
That should be the responsibility of `treeNodeService.selectVariablesTreeDataByFields`, right?

Reply:
indeed, will change